packages/jest-snapshot/src/getMatchers.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+module.exports = (filePath, options, jasmine, snapshotState) => ({
+  toMatchSnapshot: (util, customEquality) => {
+    return {
+      compare(rendered, expected) {
+
+        const specRunningFullName = snapshotState.specRunningFullName;
+
+        if (expected !== undefined) {
+          throw new Error('toMatchSnapshot() does not accepts parameters.');
+        }
+        if (typeof rendered !== 'string') {
+          throw new Error('toMatchSnapshot() only works with Strings.');
+        }
+
+        const snapshot = snapshotState.snapshot;
+
+        const specRunningCallCounter = (
+            snapshotState.specsNextCallCounter[specRunningFullName]
+        );
+
+        let pass = false;
+        let message;
+        let res = {};
+        const key = specRunningFullName + ' ' + specRunningCallCounter;
+
+        if (!snapshot.fileExists()) {
```

Comment:
what happens if the file exists but it is empty? I guess it'll work anyway because it calls `add`?

I still think we should merge `add` and `replace`. Make add not throw and remove `replace`. Then this here can be simply:

```
if (
  !snapshot.fileExists() ||
  (snapshot.has(key) && options.updateSnapshot) ||
  !snapshot.has(key)
) {
  snapshot.add(key, snapshotData);
  pass = true;
} else {
  // match and failure case
}